### PR TITLE
Recipe App Group Project

### PR DESCRIPTION
**Pull request by ezraermy**
#### This pull request implements the following features and requirements:
- Login and Registration Pages: Implemented using Devise for user authentication.
##### Food List:
- Displays a user's food items with quantities.
- Allows users to add new food items via a form.
##### Recipes List:
- Displays a user's created recipes.
- Allows the owner to delete recipes.
- Leads to recipe details.
##### Public Recipe List:
- Displays all public recipes, sorted by newest.
- Leads to recipe details.
##### Recipe Details:
- Allows recipe owners to toggle between public and private status.
- Displays recipe details for public recipes or if the user is the owner.
- Leads to a form for owners to add new food items to their recipes.
##### General Shopping List View:
- Shows missing food items for all of the user's recipes.
- Counts total food items and calculates the total price of missing items.
##### Unit and Integration Tests: Included comprehensive tests to ensure the functionality is working as expected.
Screenshot from 2023-09-07 19-02-32

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div>
+  <h2><%= @recipe.name%></h2>
+  <p><%= @recipe.preparation_time%>hours</p>
+  <p><%= @recipe.cooking_time%> hours</p>
+  <pre><%= @recipe.description%></pre>
```
Comment (elisha2kyakpo1) [inline app/views/recipes/show.html.erb]: - [ ] Kindly add the functionality for adding ingredients and generating a shopping list here (you are generating a shopping list by default). Currently, I can't make a shopping list or add ingredients. This is a requirement.


recipe-page


your-recipe-page

File: app/views/shared/_recipe.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<div class="card mx-auto my-1" style="max-width: 440px;">
+  <%= link_to recipe_path(recipe), class: "custom-link" do %>
+    <div class="card-body">
+      <h5 class="card-title"><%= recipe.name%> </h5>
+      <p class="card-text"><%= recipe.description%> </p>
+    </div>
+  <% end %>
+  <div class="text-center my-2">
+    <%= render 'shared/delete_recipe', recipe: recipe %>
```
Comment (elisha2kyakpo1) [inline app/views/shared/_recipe.html.erb]: - [ ] Although the styling is not a requirement, your designs should meet the given wireframes, kindly add some CSS to all your pages to meet this requirement.


recipe


recipe-design

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @ezraermy, @PabloEGonz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Adding and deleting items work nicely.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ezraermy** [comment]
@elisha2kyakpo1 Thanks a lot for the feedback and tips.

File: app/views/recipes/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<% @recipes.each do |recipe| %>
+  <%= render 'shared/recipe', recipe: recipe %>
+<%end%>
```
Comment (Owusu-Desmond) [inline app/views/recipes/index.html.erb:3]: - [ ] As requested by the previous reviewer, please make sure the style (layout) list of your recipes page (recipe index page) matches the given wireframe. :pray: 
  
  ### Your design
  
  image
  
  ### Given wireframe
  
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @ezraermy and @elisha2kyakpo1  , :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some changes you still need to work on before you move on to the next project but you are almost there.



 ## Highlights :+1: 
:heavy_check_mark: Good commit history.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Your project is well documented


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ezraermy** [comment]
Hi @Owusu-Desmond. We appreciate your feedback. The requested change will be implemented.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi team 👋,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️App working as expected 📌 
✔️ No linter errors 📌 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**